qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+            f"workspaces/{self.workspace_name}/providerStatus?api-version=2022-09-12-preview"
+        )
+
+        auth_headers = {"Authorization": f"Bearer {quantum_access_token}"}
+        r = self.session.get(url, headers=auth_headers)
+        devices = r.json()
+
+        devices_dict = self._helpers.device_manager(devices)
+        devices_list = [{"id": k, **v} for k, v in devices_dict.items()]
+
+        return devices_list
+
+    def get_device(self, device_id: str) -> dict[str, Any]:
+        """Get a specific Azure Quantum device."""
+        devices = self.get_devices()
+        device = None
+
+        for i in devices:
+            if i.get("id") == device_id:
+                device = i
+
+        return device
+
+    # pylint: disable=too-many-arguments
+    def create_job(
+        self, input_run: Any, name: str, provider: str, backend: str, qubits: int
+    ) -> dict[str, Any]:
+        """Create a new job on the Azure Quantum API."""
+        quantum_access_token = self._helpers.quantum_access_token(
+            self.client_id, self.client_secret, self.tenant_id, self.session
+        )
+
+        storage_access_token = self._helpers.storage_access_token(
+            self.client_id, self.client_secret, self.tenant_id, self.session
+        )
+
+        container = self._helpers.create_container(
+            storage_access_token,
+            self.subscription_id,
+            self.resource_group_name,
+            self.storage_account,
+            self.session,
+        )
+
+        job_id = container[0]
+        container_name = container[1]
+
+        job = input_run
+        routes = self._helpers.create_job_routes(
+            self.session,
+            container_name,
+            job,
+            self.connection_string,
+            self.location_name,
+            self.subscription_id,
+            self.resource_group_name,
+            self.workspace_name,
+            quantum_access_token,
+        )
+
+        containerSasUri = routes[0]
+        inputSasUri = routes[1]
+        outputSasUri = routes[2]
+        job_headers = routes[3]
+
+        job_payload = self._helpers.create_payload(
+            containerSasUri,
+            inputSasUri,
+            outputSasUri,
+            job_id,
+            job_name=name,
+            input_data_format=self.formats[provider][0],
+            provider=provider,
+            backend=self.backends[backend],
+            num_qubits=qubits,
+            num_shots=100,
+            total_count=100,
+            output_data_format=self.formats[provider][1],
+        )
+
+        job_approval = self._helpers.submit_job(
+            job_id,
+            job_payload,
+            job_headers,
+            self.location_name,
+            self.subscription_id,
+            self.resource_group_name,
+            self.workspace_name,
+            self.session,
+        )
+
+        self.jobs[job_id] = [job_headers, container_name]
```

Comment:
Using class attributes as global storage is not advisable because it can lead to unexpected behavior due to shared state among instances. Instead we want to use encapsulation to make  class's behavior more predictable i.e. Whatever data is needed for the next step should just be passed in the return.